### PR TITLE
*: Drop tectonic-system namespace

### PR DESCRIPTION
**Pull request by wking**
The kube-addon operator was the last remaining component in that namespace, and it was just controlling a metrics server.  Metrics aren't critical to cluster functions, and dropping kube-addon means we don't need the old pull secret anymore (although we will shortly need [new pull secrets for pulling private release images][1]).

[1]: https://github.com/openshift/installer/pull/663

### CODE REVIEW

**Comment by abhinavdahiya** [comment]
@wking can you also drop `role admin` and `role user` both are not required anymore.

**Comment by sjenning** [comment]
fyi `metrics-server` is being replaced with the prometheus metrics adapter anyway.  i'm looking forward to this merging! big milestone.

File: pkg/asset/manifests/template.go [hunk 1]
```diff
@@ -36,7 +36,6 @@ type bootkubeTemplateData struct {
 }
 
 type tectonicTemplateData struct {
-	KubeAddonOperatorImage string
-	PullSecret             string
-	CloudCreds             cloudCredsSecretData
+	PullSecret string
```
Comment (abhinavdahiya) [inline pkg/asset/manifests/template.go]: this can be removed

File: pkg/asset/templates/templates.go [hunk 2]
```diff
@@ -33,20 +33,15 @@ func (m *Templates) Dependencies() []asset.Asset {
 		&bootkube.KubeSystemConfigmapEtcdServingCA{},
 		&bootkube.KubeSystemConfigmapRootCA{},
 		&bootkube.KubeSystemSecretEtcdClient{},
-		&bootkube.TectonicNamespace{},
 		&bootkube.OpenshiftWebConsoleNamespace{},
 		&bootkube.OpenshiftMachineConfigOperator{},
 		&bootkube.OpenshiftClusterAPINamespace{},
 		&bootkube.OpenshiftServiceCertSignerNamespace{},
 		&bootkube.AppVersionKind{},
```
Comment (abhinavdahiya) [inline pkg/asset/templates/templates.go]: AppVersionKind also needs to be dropped.

File: pkg/asset/templates/templates.go [hunk 3]
```diff
@@ -93,20 +83,15 @@ func (m *Templates) Generate(dependencies asset.Parents) error {
 		kubeSystemConfigmapEtcdServingCA,
 		kubeSystemConfigmapRootCA,
 		kubeSystemSecretEtcdClient,
-		tectonicNamespace,
 		openshiftWebConsoleNamespace,
 		openshiftMachineConfigOperator,
 		openshiftClusterAPINamespace,
 		openshiftServiceCertSignerNamespace,
 		appVersionKind,
```
Comment (abhinavdahiya) [inline pkg/asset/templates/templates.go]: drop this

**Comment by wking** [comment]
> ... can you also drop role admin and role user both are not required anymore.

Dropped with 2ff4e5e -> af1d523.  I'm going to reap CI to get some EIP headroom too.

File: pkg/asset/manifests/template.go (hunk 1 above)
Comment (wking) [reply pkg/asset/manifests/template.go]: > this can be removed

Dropped with af1d523 -> 212450c.

File: pkg/asset/templates/templates.go (hunk 2 above)
Comment (wking) [reply pkg/asset/templates/templates.go]: > AppVersionKind also needs to be dropped.

Done with 212450c -> a73e15c.

File: pkg/asset/templates/templates.go (hunk 3 above)
Comment (wking) [reply pkg/asset/templates/templates.go]: > drop this

Done with 212450c -> a73e15c.

**Comment by wking** [comment]
Diff from my current ad6abd3 vs. #681:

```
 Gopkg.lock                                                           |   16 +--
 Gopkg.toml                                                           |    4 -
 docs/design/resource_dep.svg                                         | 1186 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++-------------------------------------------------------------------------------------
 docs/dev/libvirt-howto.md                                            |    2 +-
 pkg/asset/manifests/tectonic.go                                      |   23 +++-
 vendor/github.com/coreos/tectonic-config/LICENSE                     |  201 ---------------------------------
 vendor/github.com/coreos/tectonic-config/config/kube-addon/config.go |   23 ----
 vendor/k8s.io/apimachinery/pkg/util/rand/rand.go                     |  120 --------------------
 8 files changed, 693 insertions(+), 882 deletions(-)
```

I like the vendor, SVG, and doc differences.  I'm not sure about:

```diff
diff --git a/pkg/asset/manifests/tectonic.go b/pkg/asset/manifests/tectonic.go
index b3c067a..b90ac5a 100644
--- a/pkg/asset/manifests/tectonic.go
+++ b/pkg/asset/manifests/tectonic.go
@@ -6,6 +6,7 @@ import (
 
        "github.com/aws/aws-sdk-go/aws/session"
        "github.com/ghodss/yaml"
+       "github.com/pkg/errors"
 
        "github.com/gophercloud/utils/openstack/clientconfig"
        "github.com/openshift/installer/pkg/asset"
@@ -20,12 +21,15 @@ const (
 )
 
 var (
+       tectonicConfigPath = filepath.Join(tectonicManifestDir, "00_cluster-config.yaml")
+
        _ asset.WritableAsset = (*Tectonic)(nil)
 )
 
 // Tectonic generates the dependent resource manifests for tectonic (as against bootkube)
 type Tectonic struct {
-       FileList []*asset.File
+       TectonicConfig *configurationObject
+       FileList       []*asset.File
 }
 
 // Name returns a human friendly name for the operator
@@ -145,6 +149,21 @@ func (t *Tectonic) Load(f asset.FileFetcher) (bool, error) {
                return false, nil
        }
 
-       t.FileList = fileList
+       tectonicConfig := &configurationObject{}
+       var found bool
+       for _, file := range fileList {
+               if file.Filename == tectonicConfigPath {
+                       if err := yaml.Unmarshal(file.Data, tectonicConfig); err != nil {
+                               return false, errors.Wrapf(err, "failed to unmarshal 00_cluster-config.yaml")
+                       }
+                       found = true
+               }
+       }
+
+       if !found {
+               return false, nil
+       }
+
+       t.FileList, t.TectonicConfig = fileList, tectonicConfig
        return true, nil
 }
```

Should I drop some/all of that, and if so, why?

**Comment by abhinavdahiya** [comment]
@wking we are not setting that field anymore, so need to drop it.

**Comment by wking** [comment]
> ... we are not setting that field anymore, so need to drop it

Ok, dropped with ad6abd3 -> a14da04.  We'll see if I've released enough EIPs...

**Comment by abhinavdahiya** [comment]
i think we got all/most of what we could drop.
/lgtm

**Pull request merged**